au4/src/au3wrap/internal/processinginteraction.cpp

```diff
@@ -60,3 +60,50 @@ bool ProcessingInteraction::changeClipTitle(const processing::ClipKey& clipKey,
 
     return true;
 }
+
+bool ProcessingInteraction::removeClip(const processing::ClipKey& clipKey)
+{
+    WaveTrack* waveTrack = DomAccessor::findWaveTrack(projectRef(), TrackId(clipKey.trackId));
+    IF_ASSERT_FAILED(waveTrack) {
+        return false;
+    }
+
+    std::shared_ptr<WaveClip> clip = DomAccessor::findWaveClip(waveTrack, clipKey.index);
+    IF_ASSERT_FAILED(clip) {
+        return false;
+    }
+
+    clip->Clear(clip->Start(), clip->End());
+
+    processing::ProcessingProjectPtr prj = globalContext()->currentProcessingProject();
+    prj->onClipRemoved(DomConverter::clip(waveTrack, clip.get(), clipKey.index));
+
+    return true;
+}
+
+bool ProcessingInteraction::removeClipData(const processing::ClipKey& clipKey, double begin, double end)
+{
+    WaveTrack* waveTrack = DomAccessor::findWaveTrack(projectRef(), TrackId(clipKey.trackId));
+    IF_ASSERT_FAILED(waveTrack) {
+        return false;
+    }
+
+    std::shared_ptr<WaveClip> clip = DomAccessor::findWaveClip(waveTrack, clipKey.index);
+    IF_ASSERT_FAILED(clip) {
+        return false;
+    }
+
+    auto initialClipStart = clip->Start();
+    auto initialClipEnd = clip->End();
+
+    clip->Clear(begin, end);
+
+    processing::ProcessingProjectPtr prj = globalContext()->currentProcessingProject();
+    if (begin <= initialClipStart && end >= initialClipEnd) {
```

Comment:
doubles are not safe to compare with each other 
I suggest two options:
1. Use the `processing::secs_t` type, it is not just a double, but a wrapper over the double, for correct work with it
```
    processing::secs_t initialClipStart = clip->Start();
    processing::secs_t initialClipEnd = clip->End();

...
```

2.  Use functions from `#include "global/realfn.h"`   RealIsEqualOrMore, RealIsEqualOrLess